File: Object Oriented Programming/learning/classes_and_objects/exercises/py/ex-12.py
'''
Add two distances in inch-feet by creating a class named 'AddDistance'.
'''

# define class
class AddDistance:
    # initialize constructor
    def __init__(self, distance1, distance2):
        self.__dist1 = distance1
        self.__dist2 = distance2

    def toFeet(self):
        sum = self.__dist1 + self.__dist2
        return sum * 0.083333333
    
# create object of class
distances = AddDistance(12.0, 6.0)
print(f"Total distances in feet: {distances.toFeet()}")

File: Object Oriented Programming/learning/classes_and_objects/exercises/cpp/ex-12.cpp
/*
Add two distances in inch-feet by creating a class named 'AddDistance'.
*/
#include <iostream>
using namespace std;

// define class
class AddDistance {
    private:
        double distance1, distance2;
    
    public:
        AddDistance(double dist1, double dist2):
        distance1(dist1), distance2(dist2) {}

        double toFeet() {
            double sum = distance1 + distance2;
            return sum * 0.083333333;
        }
};

int main() {
    // create an object of class and initialize with distances
    AddDistance distances(12.0, 6.0);

    // add distances and display ther
    cout<<"Toatl distances in feet:  "<<distances.toFeet()<<endl;

    return 0;
}